src/swaps/bitcoin.rs

```diff
@@ -408,18 +409,49 @@ impl BtcSwapScript {
     ) -> Result<Vec<(OutPoint, TxOut)>, Error> {
         let electrum_client = network_config.build_client()?;
         let spk = self.to_address(network_config.network())?.script_pubkey();
-        let utxos = electrum_client.script_list_unspent(spk.as_script())?;
-        let utxo_pairs = utxos
+        let history: Vec<_> = electrum_client.script_get_history(spk.as_script())?;
+
+        let tx_heights: HashMap<_, _> = history.iter().map(|h| (h.tx_hash, h.height)).collect();
```

Comment:
We do not need the heights directly, just the information if the transaction is confirmed or not. How about having the map from tx_hash -> is_confirmed (bool)? Makes it clearer that height == 0 means unconfirmed